src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline

```diff
@@ -1030,6 +1041,89 @@ LibraryServiceApi.prototype.updateBookIndex = function(request, options, callbac
 };
 
 /**
+ * Test server streaming
+ * gRPC streaming methods don't have an HTTP equivalent and don't need to have the google.api.http option.
+ *
+ * @param {Object=} options
+ *   Optional parameters. You can override the default settings for this call, e.g, timeout,
+ *   retries, paginations, etc. See [gax.CallOptions]{@link https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the details.
+ * @returns {Stream}
+ *   An object stream which emits [StreamShelvesResponse]{@link StreamShelvesResponse} on 'data' event.
+ *
+ * @example
+ *
+ * var api = libraryV1.libraryServiceApi();
+ *
+ * api.streamShelves().on('data', function(response) {
+ *     // doThingsWith(response)
+ * });
+ */
+LibraryServiceApi.prototype.streamShelves = function(options) {
+  if (options === undefined) {
+    options = {};
+  }
+  return this._streamShelves({}, options);
+};
+
+/**
+ * Test server streaming, non-paged responses.
+ * gRPC streaming methods don't have an HTTP equivalent and don't need to have the google.api.http option.
+ *
+ * @param {Object} request
+ *   The request object that will be sent.
+ * @param {string} request.name
+ *   The name of the shelf whose books we'd like to list.
+ * @param {Object=} options
+ *   Optional parameters. You can override the default settings for this call, e.g, timeout,
+ *   retries, paginations, etc. See [gax.CallOptions]{@link https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the details.
+ * @returns {Stream}
+ *   An object stream which emits [Book]{@link Book} on 'data' event.
+ *
+ * @example
+ *
+ * var api = libraryV1.libraryServiceApi();
+ * var name = '';
+ * api.streamBooks({name: name}).on('data', function(response) {
+ *     // doThingsWith(response)
+ * });
+ */
+LibraryServiceApi.prototype.streamBooks = function(request, options) {
+  if (options === undefined) {
+    options = {};
+  }
+  return this._streamBooks(request, options);
+};
+
+/**
+ * Test bidi-streaming.
+ * gRPC streaming methods don't have an HTTP equivalent and don't need to have the google.api.http option.
+ *
+ * @param {Object=} options
+ *   Optional parameters. You can override the default settings for this call, e.g, timeout,
+ *   retries, paginations, etc. See [gax.CallOptions]{@link https://googleapis.github.io/gax-nodejs/global.html#CallOptions} for the details.
+ * @returns {Stream}
+ *   An object stream which is both readable and writable. It accepts objects
+ *   representing [DiscussBookRequest]{@link DiscussBookRequest} for write() method, and
+ *   will emit objects representing [Comment]{@link Comment} on 'data' event asynchronously.
+ *
+ * @example
+ *
+ * var api = libraryV1.libraryServiceApi();
+ *
+ * var stream = api.discussBook().on('data', function(response) {
+ *     // doThingsWith(response);
+ * });
+ * // Write request objects.
+ * stream.write({});
```

Comment:
This is just my personal preference; I think it would be more readable to our users to have request messages for writable streams be instantiated after the method call to make it clearer that the stream is a writable stream.

```js
var api = libraryV1.libraryServiceApi();

var stream = api.discussBook().on('data', function(response) {
     // doThingsWith(response);
 });

// Declare the request messages after the method is called to
// make it stand out that this stream is writable.
var comment = "";
 
// Write request objects.
stream.write({comment: comment});
```